Replace some DocumentMapper usages with MappingLookup

javanna opened the pull request:
We recently replaced some usages of DocumentMapper with MappingLookup in the search layer, as document mapper is mutable which can cause issues. In order to do that, MappingLookup grew and became quite similar to DocumentMapper in what it does and holds.

It turns out that possibly DocumentMapper does not even need to be mutable (to be verified) but in many cases we should be using MappingLookup instead of DocumentMapper, and we should even be able to remove DocumentMapper entirely in favour of MappingLookup in the long run.

This commit replaces some of the straight-forward usages.

romseygeek commented on server/src/main/java/org/elasticsearch/index/mapper/Mapping.java:
```diff
@@ -69,7 +69,7 @@ public int compare(Mapper o1, Mapper o2) {
     }
 
-    CompressedXContent toCompressedXContent() {
+    public CompressedXContent toCompressedXContent() {
```
Can we add javadocs if we're making it public?

romseygeek commented on server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java:
```diff
@@ -339,7 +339,7 @@ public CacheKey cacheKey() {
     }
 
-    Mapping getMapping() {
+    public Mapping getMapping() {
```
Same here

romseygeek reviewed (approved):
LGTM, modulo a couple of nits re javadoc

javanna committed:
Replace some DocumentMapper usages with MappingLookup

We recently replaced some usages of DocumentMapper with MappingLookup in the search layer, as document mapper is mutable which can cause issues. In order to do that, MappingLookup grew and became quite similar to DocumentMapper in what it does and holds.

In many cases it makes sense to use MappingLookup instead of DocumentMapper, and we may even be able to remove DocumentMapper entirely in favour of MappingLookup in the long run.

This commit replaces some of its straight-forward usages.

javanna commented:
I pushed an update, @romseygeek you may want to have another look just to make sure :)

javanna committed:
spotless

javanna committed:
One more

romseygeek reviewed (approved):
LGTM

javanna committed:
iter

javanna committed:
address tests
